src/main/java/org/cryptomator/common/LazyProcessedProperties.java

```diff
@@ -0,0 +1,73 @@
+package org.cryptomator.common;
+
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.Properties;
+import java.util.regex.Pattern;
+
+public class LazyProcessedProperties extends PropertiesDecorator {
+
+	private static final Pattern TEMPLATE = Pattern.compile("@\\{(\\w+)}");
+
+	private final Map<String, String> env;
+
+	public LazyProcessedProperties(Properties props, Map<String, String> systemEnvironment) {
+		super(props);
+		this.env = systemEnvironment;
+	}
+
+	@Override
+	public String getProperty(String key) {
+		var value = delegate.getProperty(key);
+		if (key.startsWith("cryptomator.") && value != null) {
+			return process(value);
+		} else {
+			return value;
+		}
+	}
+
+	@Override
+	public String getProperty(String key, String defaultValue) {
+		var value = delegate.getProperty(key, defaultValue);
+		if (key.startsWith("cryptomator.") && value != null) {
+			return process(value);
+		} else {
+			return value;
+		}
+	}
```

Comment:
please add tests with 100% branch coverage (i.e. both `cryptomator.*` as well as `example.*` keys as well as value present and absent)

Reply:
(while trivial enough to see that it will not cause problems, we need to protect against regressions, as this may affect the whole application startup sequence)

Reply:
done in https://github.com/cryptomator/cryptomator/pull/2957/commits/cee0486d71037233b30672d0b5341b7e6417b3cb